src/Marten/Services/CommandRunnerExtensions.cs

```diff
@@ -29,7 +29,7 @@ public static QueryPlan ExplainQuery(this IManagedConnection runner, NpgsqlComma
             cmd.CommandText = string.Concat($"explain ({config} format json) ", cmd.CommandText);
             using (var reader = runner.ExecuteReader(cmd))
             {
-                var queryPlans = reader.Read() ? serializer.FromJson<QueryPlanContainer[]>(reader.GetTextReader(0)) : null;
+                var queryPlans = reader.Read() ? serializer.FromJson<QueryPlanContainer[]>(reader.GetStream(0)) : null;
```

Comment:
One thing TODO here might be to make the ISerializer interface take in the DbDataReader and index, then let each serializer to whatever they need to do. So you don't have to make the serializers conform to something that isn't idea.

Reply:
Refactored in https://github.com/JasperFx/marten/pull/1685/commits/be80fc6c0b93a4b901f3984ac1e8b9df31b94789.